app/views/layouts/_adv_search_body.html.haml

```diff
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:search_type] == "global"
```

Comment:
.. so, changing the checkbox calls `adv_search_name_typed`, which sets `@edit[:search_type]` correctly, looks like we're fine there...

`adv_search_button` also sets it on load (`loadit`) (and `saveit` respects the value (and only that value!) and saves) .. **but** `loadit` never happens when just editing the current filter.

It also looks like `@edit[:expression][:selected]` usually only contains an `:id` and `:description`, not `:typ` - so the old code is probably wrong...

The new code is also wrong in that it no longer makes sense to do the whole condition (`@a && @a[:b] && @a[:b][:c]` makes sense, but `@a && @a[:b] && @d` doesn't ;) ) - sounds like `- checked = @edit && @edit[:search_type] == "global"` is all you need.

But.. the change is *probably* right, given that that's the value we use for saving - but we also need to set `@edit[:search_type]` when starting to edit the search. ... But from where? If we look at `@edit[:expression][:selected]` , it has a value of `"default"`, not `"global"`...

... ahh, because "default" is global, for some reason, and "user" is non-global.

So.. in my DB:
default	110
global	8
user	6

So clearly we have a 3-state here, which makes it much less surprising the checkbox doesn't work properly :).

What makes it work for me is this:

```diff
--- a/app/controllers/application_controller/filter.rb
+++ b/app/controllers/application_controller/filter.rb
@@ -157,6 +157,10 @@ module ApplicationController::Filter
     # Rebuild the pulldowns if opening the search box
     @edit[@expkey].prefill_val_types unless @edit[:adv_search_open]
 
+    unless @edit[:adv_search_open]
+      @edit[:search_type] = @edit[:expression][:selected][:typ] if @edit.fetch_path(:expression, :selected, :typ).present?
+    end
+
     render :update do |page|
       page << javascript_prologue
       if @edit[:adv_search_open] == true
```

together with

```diff
--- a/app/views/layouts/_adv_search_body.html.haml
+++ b/app/views/layouts/_adv_search_body.html.haml
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && %w(global default).include?(@edit[:search_type])
               = check_box_tag("search_type", "1", checked,
                 :style                      => "width: 20px",
                 "data-miq_observe_checkbox" => {:url => url2}.to_json)
```

but I'm *very* unsure that `default` always `== global`.

@h-kataria any idea why there's 3 different values for `search_type` in MiqSearch?